### DISCUSSION TITLE
Can ephem easily determine local timezone?

### DISCUSSION THREAD

**Discussion by kovar-ursa** [brandon-rhodes/pyephem#226] [General]
I use ephem to determine local twilight for a large number of points. 

I also need to determine the timezone for these points. To do so, I currently use `tzwhere` but it is a) unsupported and b) a bit flakey. 

If I could calculate both time of day and timezone with one function using ephem, that would be a huge win.

**Comment by brandon-rhodes** [comment]
Alas, no: that would be a question for a geography library, which would have to have a map of the time zones across the entire planet. A search for "python translate latitude longitude into time zone" brings up:

https://stackoverflow.com/questions/15742045/getting-time-zone-from-lat-long-coordinates
https://stackoverflow.com/questions/16086962/how-to-get-a-time-zone-from-a-location-using-latitude-and-longitude-coordinates/16086964

It looks like `pytzwhere` might still be state-of-the-art, though as you note it doesn't seem to have been touched for a very long time. It might need a new maintainer if issues are piling up that need to be addressed?

**Reply by kovar-ursa** [reply]
I think `pytzwhere` will do for the moment. If I can find and fix the minor issues I'm finding I'll consider forking it.

Thank you!
